cmd/juju/application/deploy.go

```diff
@@ -229,6 +233,14 @@ func NewDeployCommandWithAPI(newAPIRoot NewAPIRootFn) cmd.Command {
 	return modelcmd.Wrap(cmd)
 }
 
+// NewDeployCommandWithSteps returns a command to deploy services with custom deploy steps.
+func NewDeployCommandWithSteps(steps []DeployStep) cmd.Command {
```

Comment:
NewDeployCommandWithSteps(steps ...DeployStep) .. that way we can call NewDeployCommandWithSteps() in tests